common/src/api/external/error.rs

```diff
@@ -300,6 +306,15 @@ impl Error {
         Error::Conflict { message: MessagePair::new(message.into()) }
     }
 
+    /// Generates an [`Error::NotFound`] with a specific message.
+    ///
+    /// This is used in cases where a generic 404 is required. For cases where
+    /// there is a ResourceType, use a function that produces
+    /// [`Error::ObjectNotFound`] instead.
+    pub fn not_found(message: impl Into<String>) -> Error {
```

Comment:
How picky/careful do we want to be about trying to help avoid misuse of this? E.g., would it make sense to name this something more annoying like `non_resource_not_found` to make it stand out at call sites?

Reply:
agreed, done in 414298e89